index.js

```diff
@@ -41,19 +42,101 @@ class ScmBase {
-     * @param {String}    scmUrl        Scm Url to format properly
+     * Parse the url for a repo for the specific source control
+     * @method parseurl
+     * @param  {Object}    config
+     * @param  {String}    config.checkoutUrl       Url to parse
+     * @param  {String}    config.token             The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    parseUrl(config) {
+        return validate(config, dataSchema.plugins.scm.parseUrl)
+            .then(validUrl => this._parseUrl(validUrl))
+            .then(uri => validate(uri, dataSchema.models.pipeline.base.scmUri));
+    }
+
+    _parseUrl() {
+        return Promise.reject('Not implemented');
+    }
+
+    /**
+     * Parse the webhook for the specific source control
+     * @method parseHook
+     * @param  {Object}     headers     The request headers associated with the webhook payload
+     * @param  {Object}     payload     The webhook payload received from the SCM service
+     * @return {Object}                 A key-map of data related to the received payload
+     */
+    parseHook(headers, payload) {
+        const result = this._parseHook(headers, payload);
+
+        return validate(result, dataSchema.core.scm.hook);
+    }
+
+    _parseHook() {
+        throw new Error('Not implemented');
+    }
+
+    /**
+     * Decorate the url for the specific source control
+     * @method decorateUrl
+     * @param  {Object}    config
+     * @param  {String}    config.scmUri       SCM uri to decorate
+     * @param  {String}    config.token        The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    decorateUrl(config) {
+        return validate(config, dataSchema.plugins.scm.decorateUrl)
+            .then(validUrl => this._decorateUrl(validUrl))
+            .then(decoratedUrl => validate(decoratedUrl, dataSchema.core.scm.repo));
+    }
+
+    _decorateUrl() {
+        return Promise.reject('Not implemented');
+    }
+
+    /**
+     * Decorate the commit for the specific source control
+     * @method decorateCommit
+     * @param  {Object}    config
+     * @param  {String}    config.sha           Commit sha to decorate
+     * @param  {String}    config.scmUri        SCM uri
+     * @param  {String}    config.token         The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    decorateCommit(config) {
+        return validate(config, dataSchema.plugins.scm.decorateCommit)
+            .then(validCommit => this._decorateCommit(validCommit))
+            .then(decoratedCommit => validate(decoratedCommit, dataSchema.core.scm.commit));
+    }
+
+    _decorateCommit() {
+        return Promise.reject('Not implemented');
+    }
+
+    /**
+     * Decorate the author for the specific source control
+     * @method decorateAuthor
+     * @param  {Object}    config
+     * @param  {String}    config.username  Author to decorate
+     * @param  {String}    config.scmUri    SCM uri
```

Comment:
Removing `scmuri` as param for `decorateAuthor`

We will eventually pass in some sort of `userUri` in the form of `github.com:stjohnjohnson`, but will save that refactoring for a different pull request